configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -494,7 +494,11 @@ public void addConfigurationBindingListener(ConfigurationBindingListener listene
                     operativeName = fullName;
                 }
                 else if (!value.equals(operativeValue)) {
-                    problems.addError("Value for property '%s' (=%s) conflicts with property '%s' (=%s)", fullName, value, operativeName, operativeValue);
+                    if (attribute.isSecuritySensitive()) {
+                        problems.addError("Value for property '%s' (=[REDACTED]) conflicts with property '%s' (=[REDACTED])", fullName, operativeName);
+                    } else {
+                        problems.addError("Value for property '%s' (=%s) conflicts with property '%s' (=%s)", fullName, value, operativeName, operativeValue);
+                    }
```

Comment:
I'm not a fan of the duplication here, which is why the original code used the `printableValue` approach.